From f262291ec644752c8fc7e1598fafd568e5302a78 Mon Sep 17 00:00:00 2001
From: Lyon Wang <lyon.wang@nxp.com>
Date: Thu, 1 Feb 2018 11:37:34 +0800
Subject: [PATCH 08/68] Update for gplay not set video-sink by default

MMFMWK-7883
- Add video-direction property in set-rotate
- Update get video-sink for autoplug video-sink case
- Add kmssink priority logic
- kmssink include gstimxcommon.h

Upstream-Status: Pending
Signed-off-by: Lyon Wang <lyon.wang@nxp.com>

(cherry picked from commit 39c98b4abcacae9757e19252baa26f5a619b6b22)

This reverts commit 89b9bfa84c419acadbfa2ca06cc05af0d12b1dae.
kmssink: don't raise rank of kmssink
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst-libs/gst/play/gstplay.c | 8 ++++++++
 sys/kms/gstkmssink.c        | 1 +
 2 files changed, 9 insertions(+)

diff --git a/gst-libs/gst/play/gstplay.c b/gst-libs/gst/play/gstplay.c
index 0d417df..a8b65f3 100644
--- a/gst-libs/gst/play/gstplay.c
+++ b/gst-libs/gst/play/gstplay.c
@@ -4733,6 +4733,8 @@ gst_play_get_video_sink (GstPlay * self)
     GST_WARNING_OBJECT (self, "No video-sink found");
     return NULL;
   }
+  if (TRUE != GST_IS_BIN((GstBin*) sink))
+     return sink;
   it = gst_bin_iterate_sinks ((GstBin *) sink);
   do {
     rc = gst_iterator_next (it, &item);
@@ -4784,6 +4786,9 @@ gst_play_set_rotate (GstPlay * self, gint rotation)
     g_object_set (G_OBJECT (video_sink), "reconfig", 1, NULL);
   } else if (g_object_class_find_property (gobjclass, "rotate-method")) {
     g_object_set (G_OBJECT (video_sink), "rotate-method", rotation / 90, NULL);
+  }  else if (g_object_class_find_property (gobjclass, "video-direction")) {
+    g_object_set (G_OBJECT (video_sink), "video-direction", rotation / 90, NULL);
+    g_object_set (G_OBJECT (video_sink), "reconfig", 1, NULL);
   } else {
     GST_INFO_OBJECT (self, "can't set rotation for current video sink %s'",
         gst_element_get_name (video_sink));
@@ -4821,6 +4826,9 @@ gst_play_get_rotate (GstPlay * self)
   } else if (g_object_class_find_property (gobjclass, "rotate-method")) {
     g_object_get (G_OBJECT (video_sink), "rotate-method", &rotation, NULL);
     rotation = rotation * 90;
+  }   else if (g_object_class_find_property (gobjclass, "video-direction")) {
+    g_object_get (G_OBJECT (video_sink), "video-direction", &rotation, NULL);
+    rotation = rotation * 90;
   }
 
   GST_DEBUG_OBJECT (self, "get rotation degree '%d'", rotation);
diff --git a/sys/kms/gstkmssink.c b/sys/kms/gstkmssink.c
index 3b4711f..7d2dfd9 100644
--- a/sys/kms/gstkmssink.c
+++ b/sys/kms/gstkmssink.c
@@ -62,6 +62,7 @@
 #include "gstkmsutils.h"
 #include "gstkmsbufferpool.h"
 #include "gstkmsallocator.h"
+#include "gstimxcommon.h"
 
 #ifdef HAVE_DRM_HDR
 #include <math.h>
-- 
2.43.0

